packages/sls-env/src/index.ts

```diff
@@ -280,6 +296,30 @@ const environment = <H extends Handler<any, any, any>, C, D, P = HandlerPayload<
             logger.trace({ payload }, 'created payload')
             return { logger, config, dependencies, payload, invocationId }
           })
+          // override event based dependencies
+          .then(({ logger, config, dependencies: _dependencies, payload, invocationId }) => {
+            logger.trace('about to resolve event based dependencies')
+            // need to shallow clone to ensure reference to cached applicationDependencies is not updated
+            const dependencies = { ..._dependencies }
+            Object.entries(applicationDependencies).forEach(([dependencyName, dependency]) => {
```

Comment:
favour immutability over mutation `reduce` would be a better option then `forEach`